kview/src/main/java/dev/ikm/komet/kview/mvvm/model/DataViewModelHelper.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2015 Integrated Knowledge Management (support@ikm.dev)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.ikm.komet.kview.mvvm.model;
+
+import dev.ikm.komet.framework.view.ViewProperties;
+import dev.ikm.tinkar.common.id.IntIdSet;
+import dev.ikm.tinkar.common.id.PublicId;
+import dev.ikm.tinkar.entity.ConceptEntity;
+import dev.ikm.tinkar.entity.Entity;
+import dev.ikm.tinkar.entity.EntityService;
+import dev.ikm.tinkar.terms.EntityFacade;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static dev.ikm.tinkar.terms.TinkarTerm.*;
+
+public class DataViewModelHelper {
+
+    public static final Set<EntityFacade> DATA_TYPE_OPTIONS = Collections.unmodifiableSet(Set.of(
+            STRING,
+            COMPONENT_FIELD,
+            COMPONENT_ID_SET_FIELD,
+            COMPONENT_ID_LIST_FIELD,
+            DITREE_FIELD,
+            DIGRAPH_FIELD,
+            CONCEPT_FIELD,
+            SEMANTIC_FIELD_TYPE,
+            INTEGER_FIELD,
+            FLOAT_FIELD,
+            BOOLEAN_FIELD,
+            BYTE_ARRAY_FIELD,
+            ARRAY_FIELD,
+            INSTANT_LITERAL,
+            LONG
+    ));
```

Comment:
@swaroopsalvi 
Can we make these **static variables** into **methods** instead of a variable name? 
**e.g.**
```java
public static Set<ConceptEntity> fetchFieldDefinitionDataTypes(ViewProperties viewProperties, PublicId publicId) {
    return Set.of(
            STRING,
            COMPONENT_FIELD,
            COMPONENT_ID_SET_FIELD,
            COMPONENT_ID_LIST_FIELD,
            DITREE_FIELD,
            DIGRAPH_FIELD,
            CONCEPT_FIELD,
            SEMANTIC_FIELD_TYPE,
            INTEGER_FIELD,
            FLOAT_FIELD,
            BOOLEAN_FIELD,
            BYTE_ARRAY_FIELD,
            ARRAY_FIELD,
            INSTANT_LITERAL,
            LONG
    );
}
```

It was my mistake on suggesting a **variable** at the time of the first review. Binding objects are connected to the database because they have nids. 

Let me explain. If a caller is not connected to the (entity) database you will get the following:
```java
Exception in thread "main" java.lang.ExceptionInInitializerError
	at dev.ikm.komet.application.test/dev.ikm.komet.app.test.ConceptViewModelTest2.main(ConceptViewModelTest2.java:36)
Caused by: java.lang.IllegalStateException: No provider. Call Select provider prior to get()
	at dev.ikm.tinkar.common@1.58.0/dev.ikm.tinkar.common.service.PrimitiveData.get(PrimitiveData.java:234)
	at dev.ikm.tinkar.terms@1.58.0/dev.ikm.tinkar.terms.EntityProxy.nid(EntityProxy.java:141)
```

While the caller will still get a similar error when calling the function at least they can handle exceptions. The caller of a variable would be faced with the entire class `DataModelHelper` not class loaded correctly. There are too many stateful objects that need to be 'realized or hydrated' meaning used as public static final variables. This is an anti-pattern that test code will have a hard time dealing with. We may rethink about using ProtoBuffer objects instead of EntityFacades in situations passing DTOs.

Reply:
Done

Reply:
@swaroopsalvi Carl asked to rename the file DataModelHelper?  just take the word 'view' out of the name.

Reply:
Done

Reply:
@dholubek @carldea Let me know if you see anything else